app/grandchallenge/evaluation/models.py

```diff
@@ -582,9 +591,49 @@ def has_pending_evaluations(self, *, user):
             .exists()
         )
 
+    @property
+    def submission_period_open(self):
+        now = timezone.now()
+        if self.submissions_open and self.submissions_close:
+            submission_period_open = (
+                now > self.submissions_open and now < self.submissions_close
+            )
+        elif self.submissions_open:
+            submission_period_open = now > self.submissions_open
+        elif self.submissions_close:
+            submission_period_open = now < self.submissions_close
+        else:
+            submission_period_open = True
+        return submission_period_open
+
     @property
     def open_for_submissions(self):
-        return self.submission_limit != 0
+        if self.submission_limit == 0:
+            return False
+        elif not self.submission_period_open:
+            return False
+        else:
+            return True
```

Comment:
I think this can all simplify to `return self.submission_period_open and self.submission_limit > 0`